lib/if.c

```diff
@@ -111,15 +113,27 @@ int if_cmp_name_func(const char *p1, const char *p2)
 		if (!*p2)
 			return 1;
 
-		x1 = strtol(p1, (char **)&p1, 10);
-		x2 = strtol(p2, (char **)&p2, 10);
+		x1 = strtol(p1, (char **)&tmp1, 10);
+		x2 = strtol(p2, (char **)&tmp2, 10);
 
 		/* let's compare numbers now */
 		if (x1 < x2)
 			return -1;
 		if (x1 > x2)
 			return 1;
 
+		/* Compare string if numbers are equal (distinguish foo-1 from
+		 * foo-001)
+		 */
+		l1 = strspn(p1, "0123456789");
+		l2 = strspn(p2, "0123456789");
+		if (l1 != l2)
+			return (strcmp(p1, p2));
```

Comment:
`strcmp` will sort `foo001` before `foo1`. I think it's wrong.
How about comparing `l1` and `l2` instead?

Reply:
Why is it wrong? Which rules are you referring to?

Reply:
Why should you compare l1 and l2 ?! What are you expecting ?

+1 with @NicolasDichtel

Reply:
@NicolasDichtel "wrong" may be a bad choice of words here. I wanted to say that it's not expected, at least to me. I think shorter names should be sorted before longer ones:
```
foo1
foo01
foo0001
```
and `strcmp` will do it in reverse. What do you think?

@vjardin `l1` and `l2` are lengths of number parts and their difference is the number of preceding zeroes in the longer name. Using the difference is faster than `strcmp` and allows to make the order that I explain above.

Reply:
And we can go back to my initial comment: why is this function needed at all? I didn't make any bench, but it seems obvious that replacing if_cmp_name_func() with strcmp() will be much faster. Why should this function orders 'foo2' before 'foo10', why is this necessary?

Reply:
It's just a common practice for networking hardware to sort the interfaces in such order in running-config and "show" commands.

Reply:
Is it worst doing like this? If we want an optimized function, is it not better to use standard functions like strcmp()?

Reply:
> @NicolasDichtel "wrong" may be a bad choice of words here. I wanted to say that it's not expected, at least to me. I think shorter names should be sorted before longer ones:
> 
> ```
> foo1
> foo01
> foo0001
> ```
> 
> and `strcmp` will do it in reverse. What do you think?
It seems more common to order foo01 before foo1:
$ echo -e "foo1\nfoo01\nfoo0001" | sort -n
foo0001
foo01
foo1